test/e2e/etcd_failure.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const etcdFailureDuration = 20 * time.Second
+
+type etcdFailureCase struct {
+	c  *client.Client
+	ns string
+}
+
+var _ = Describe("Etcd failure", func() {
+	defer GinkgoRecover()
+
+	var c *client.Client
+	var ns string
+
+	BeforeEach(func() {
+		By("preparing namespace")
+
+		ns = "e2e-test-" + string(util.NewUUID())
+		var err error
+
+		c, err = loadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		Logf("creating new namespace")
+		_, err = createNamespaceIfDoesNotExist(c, ns)
+		Expect(err).NotTo(HaveOccurred())
+
+		Logf("creating replication controller before failure")
+		Expect(RunRC(RCConfig{
+			Client:    c,
+			Name:      "baz",
+			Namespace: ns,
+			Image:     "nginx",
+			Replicas:  1,
+		})).NotTo(HaveOccurred())
+	})
+
+	It("should recover from network partition with master", func() {
+		var dropEtcdPackets sync.Once
```

Comment:
It's not clear to me why you need this?  Under what conditions could it be called more than once?